lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -214,6 +215,72 @@ export function useAPIFetch<T = unknown>(
   return useFetch(path ? `${path}${paramStr}` : null, fetcher);
 }
 
+export type PolledAPIFetchOptions<T> = {
+  /** Path for API call */
+  path: string;
+  /** Query params for API call */
+  params?: QueryParams;
+  /** Determines if, based on the result, the API should be called again */
+  shouldRefresh: (result: FetchResult<T>) => boolean;
+
+  /**
+   * Amount of ms after which a refresh should happen, if shouldRefresh()
+   * returns true.
+   * Defaults to 500ms.
+   */
+  refreshAfter?: number;
+
+  /** Test seam */
+  _setTimeout?: typeof setTimeout;
+  /** Test seam */
+  _clearTimeout?: typeof clearTimeout;
+};
+
+/**
+ * Hook that fetches data using authenticated API requests, allowing it to be
+ * refreshed periodically.
+ */
+export function usePolledAPIFetch<T = unknown>({
```

Comment:
I think it would be useful here to say something about the intended use case. Something like:

> Hook that fetches data using authenticated API requests.
> 
> This is a variant of {@link useAPIFetch} that supports automatically refreshing results at an interval until some condition is met. This is useful for example when calling an API that reports the status of a long-running operation.